### DISCUSSION TITLE
Does llama.cpp support CUDA through HIP?

### DISCUSSION THREAD

**Discussion by MarioIshac** [ggml-org/llama.cpp#10892] [General]
I am already running llama.cpp using HIP for ROCm and it is great, but had the above question out of curiosity. If HIP is a [thin wrapper of CUDA](https://rocm.docs.amd.com/projects/HIP/en/docs-6.0.0/user_guide/hip_porting_guide.html#general-tips) on CUDA machines, does that mean that following [this AMD guide](https://github.com/ggerganov/llama.cpp/blob/master/docs/build.md#hip) would produce a `llama-server` / artifact that would be GPU accelerated on an Nvidia machine?

**Comment by IMbackK** [comment]
sure yeah that should work, i dont think anyone has tried however
